### ISSUE TITLE
Add option to undo a generate command

### ISSUE THREAD

**Issue by jotterkain** [nestjs/nest-cli#1915]
### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

It would be pretty cool if we had an option to undo the changes made by `nest generate` command.
This features request has already been made #496.
But i didn't like how the team handled it. 
It's not because of `--dry-run` that this feature does't worth an issue here.

### Describe the solution you'd like

After running `nest genereate module`, i want to be able to do something like `nest remove module`.

### Teachability, documentation, adoption, migration strategy

After running `nest genereate module`, i want to be able to do something like `nest remove module`.

### What is the motivation / use case for changing the behavior?

This will help us to not look for every single change made in project to delete them.  With a single command, they are undo.

**Comment by micalevisk** [comment]
but 'undo' is not the same as 'remove'

Undo is hard. Remove is not that much.

What are you proposing here, exactly? 🤔 

I didn't get why the dry-run mode isn't enough tho.

**Comment by jotterkain** [comment]
Remove, Undo, i mean we should be able to delete generated files with one command. I know it's not easy. I'm just praying the core team to look into it. it would be nice. However, if it is not relevant. then okay.

**Comment by jmcdo29** [comment]
You could theoretically make your own schematic to do this. It would look weird but it could be done. `nest generate -c custom-collection remove <name>` and it could search through your project to find files that match the `<name>`. Along with removing the files it would need to update the `app.module.ts` file as well.

**Comment by micalevisk** [comment]
@jotterkain 
> I know it's not easy. I'm just praying the core team to look into it. 

it was already in 2021 like you've mentioned.

---

To me, the only scenario on which that command would be useful is the one on which the CLI should revert everything. Did you got another one?

So, for the following cases (& more), `undo` shouldn't be used:

- `generate` but I don't want few files
- `generate` but notice some typo in the resource name after editing some of those generated/updated files

Basically, Git or other versioning tool is likely to be more useful than `undo`. Thus, we end up having a piece of code that is hard to maintain just because of one rare(?) case :/ I believe that Angular schematics doesn't has a revert abstraction, that's why this would be hard to implement.

**Comment by kamilmysliwiec** [comment]
Thanks for your suggestion!

There are no plans to implement it in the foreseeable future.

If you think your request could live outside Nest's scope, we'd encourage you to collaborate with the [community](https://discord.gg/G7Qnnhy) on publishing it as an open source package.
